### PR TITLE
Tidy things up before publishing

### PR DESCRIPTION
**Pull request by ostefano**

### CODE REVIEW

File: .github/FUNDING.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # These are supported funding model platforms
 
-github: [coolacid]
+github: [ostefano]
```
Comment (Phat3) [inline .github/FUNDING.yml]: Shouldn't we remove the whole file?

File: README.md [hunk 2]
```diff
@@ -17,14 +15,28 @@ This is based on some of the work from the DSCO docker build, nearly all of the
...
+Nevertheless the underlying spirit of this fork is to allow "repeatable deployments", and all pull requests in this direction will be merged.
 
--   MISP version is the MISP tag we're building
--   Our build version is the iteration for our changes with the same MISP version
+## Versioning
+
+GitHub builds the images automatically and pushes them to [Docker hub](https://hub.docker.com/r/ostefano/misp-docker). We do not use tags and versioning works as follows:
+
+-   MISP (and modules) version specified inside the `template.env` file
+-   Docker images are tagged based on the commit hash
```
Comment (Phat3) [inline README.md:39]: We should specify that `core` and `modules` are tagged as `core-commit-sha1[0:7]` and `modules-commit-sha1[0:7]` respectively. The latest images would still have the additional tags `core-latest` and `modules-latest`

File: build-docker-compose.yml [hunk 3]
```diff
@@ -15,3 +15,4 @@ services:
         context: modules/.
         args:
             - MODULES_TAG=${MODULES_TAG}
+            - MODULES_COMMIT=${MODULES_COMMIT}
```
Comment (Phat3) [inline build-docker-compose.yml]: This variable is required but it is commented out in `template.env`. Does it have to be mandatory?

File: docker-compose.yml [hunk 4]
```diff
@@ -41,23 +41,23 @@ services:
       - "./logs/:/var/www/MISP/app/tmp/logs/:delegated"
       - "./files/:/var/www/MISP/app/files/:delegated"
       - "./ssl/:/etc/nginx/certs/:delegated"
+      - "./gnupg/:/var/www/MISP/.gnupg/:delegated"
```
Comment (Phat3) [inline docker-compose.yml]: `ansible-environments` should be updated as well to reflect these changes

File: docker-compose.yml [hunk 5]
```diff
@@ -41,23 +41,23 @@ services:
...
 #      - "./examples/custom-entrypoint.sh:/custom-entrypoint.sh" # Use the example custom-entrypoint.sh
-      - "./.gnupg/:/var/www/MISP/.gnupg/:delegated"
     environment:
       - "HOSTNAME=https://localhost"
       - "REDIS_FQDN=redis"
       - "INIT=true"             # Initialze MISP, things includes, attempting to import SQL and the Files DIR
       - "CRON_USER_ID=1"        # The MISP user ID to run cron jobs as
       # Synchronization Servers settings
-      - "SYNCSERVERS=1"
+      - "SYNCSERVERS=${SYNCSERVERS}"
```
Comment (Phat3) [inline docker-compose.yml:59]: ansible-environments should be updated as well to reflect these changes

File: docker-compose.yml [hunk 6]
```diff
@@ -41,23 +41,23 @@ services:
...
       - "SYNCSERVERS_1_NAME=${SYNCSERVERS_1_NAME}"
       - "SYNCSERVERS_1_UUID=${SYNCSERVERS_1_UUID}"
       - "SYNCSERVERS_1_KEY=${SYNCSERVERS_1_KEY}"
       - |
         SYNCSERVERS_1_DATA=
         {
-          "url": "https://intel.thedfirreport.com/",
+          "url": "${SYNCSERVERS_1_URL}",
```
Comment (Phat3) [inline docker-compose.yml:66]: ansible-environments should be updated as well to reflect these changes

**Review by Phat3 — CHANGES_REQUESTED** [review]

**Comment by Phat3** [comment]
Can we also remove all the parts that are commented out in `docker-compose.yml`? I also think that `build-docker-compose.yml` can be removed and merged inside `docker-compose.yml`.

File: .github/FUNDING.yml (hunk 1 above)
Comment (ostefano) [reply .github/FUNDING.yml]: And destroy my dreams to become wealthy?

ACK, will remove.

File: README.md (hunk 2 above)
Comment (ostefano) [reply README.md:39]: ACK

File: build-docker-compose.yml (hunk 3 above)
Comment (ostefano) [reply build-docker-compose.yml]: Ideally the dockerfile check whether it's empty or not, if commit is not empty, it overrides tag.

File: docker-compose.yml (hunk 4 above)
Comment (ostefano) [reply docker-compose.yml]: ACK

File: docker-compose.yml (hunk 5 above)
Comment (ostefano) [reply docker-compose.yml:59]: ACK

File: docker-compose.yml (hunk 6 above)
Comment (ostefano) [reply docker-compose.yml:66]: ACK

**Comment by ostefano** [comment]
Merging would be nice. Will work on it, and re-submit this one and open a twin one for ansible-env

File: .github/FUNDING.yml (hunk 1 above)
Comment (Phat3) [reply .github/FUNDING.yml]: "OnlyFans" only as funding :joy:

File: build-docker-compose.yml (hunk 3 above)
Comment (Phat3) [reply build-docker-compose.yml]: But docker itself would complain. You can add a default value with the notation `${MODULES_COMMIT:-<DefaultValue>}`

**Review by Phat3 — APPROVED** [review]

File: build-docker-compose.yml (hunk 3 above)
Comment (ostefano) [reply build-docker-compose.yml]: @Phat3 I am not sure whether this is he case. Look at the Dockerfile inside `server`. We do the same thing and we have no side-effect.

Comment (ostefano) [reply build-docker-compose.yml]: I get this when building:
```
WARN[0000] The "MISP_COMMIT" variable is not set. Defaulting to a blank string.
WARN[0000] The "MODULES_COMMIT" variable is not set. Defaulting to a blank string.
```

Comment (Phat3) [reply build-docker-compose.yml]: Yep you get warnings though. If you are fine with those we can leave it as is

File: server/files/entrypoint_nginx.sh [hunk 7]
```diff
@@ -145,13 +144,21 @@ echo "Configure MISP | Initialize misp base config..." && init_misp_config
 echo "Configure MISP | Sync app files..." && sync_files
 
 echo "Configure MISP | Enforce permissions ..."
-echo "... chown -R www-data:www-data /var/www/MISP ..." && find /var/www/MISP -not -user www-data -exec chown www-data:www-data {} +
-echo "... chmod -R 0750 /var/www/MISP ..." && find /var/www/MISP -perm 550 -type f -exec chmod 0550 {} + && find /var/www/MISP -perm 770 -type d -exec chmod 0770 {} +
-echo "... chmod -R g+ws /var/www/MISP/app/tmp ..." && chmod -R g+ws /var/www/MISP/app/tmp
-echo "... chmod -R g+ws /var/www/MISP/app/files ..." && chmod -R g+ws /var/www/MISP/app/files
-echo "... chmod -R g+ws /var/www/MISP/app/files/scripts/tmp ..." && chmod -R g+ws /var/www/MISP/app/files/scripts/tmp
```
Comment (Phat3) [inline server/files/entrypoint_nginx.sh:152]: Is it correct that this folder is not modified anymore?

File: server/files/entrypoint_internal.sh [hunk 8]
```diff
@@ -214,6 +214,13 @@ get_server() {
      -H "Content-type: application/json" ${HOSTNAME}/servers | jq -e -r ".[] | select(.Server[\"name\"] == \"${1}\") | .Server.id"
 }
 
+updateComponents() {
```
Comment (Phat3) [inline server/files/entrypoint_internal.sh:217]: Is this a new functionality?

File: server/Dockerfile [hunk 9]
```diff
@@ -137,6 +137,8 @@ ARG PHP_VER
     RUN if [ ! -z ${MISP_COMMIT} ]; then cd /var/www/MISP && git checkout ${MISP_COMMIT}; fi; \
         # We build the MISP modules outside, so we don't need to grab those submodules
         cd /var/www/MISP/app || exit; git submodule update --init --recursive .;
+    # Remove some old and broken links that pollute the log files
+    RUN rm -rf /var/www/MISP/INSTALL/old
```
Comment (Phat3) [inline server/Dockerfile]: Can we inline in the `RUN` above so we don't create an additional layer?

**Review by Phat3 — APPROVED** [review]

File: docker-compose.yml [hunk 10]
```diff
@@ -30,34 +30,40 @@ services:
...
           "pull_rules": "{\"tags\":{\"OR\":[],\"NOT\":[]},\"orgs\":{\"OR\":[],\"NOT\":[]},\"url_params\":\"{\\\"searchanalysis\\\": \\\"2\\\"}\"}",
           "pull": true
         }
```
Comment (Phat3) [inline docker-compose.yml:69]: Can we remove also all those comments below

**Review by Phat3 — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 10 above)
Comment (Phat3) [reply docker-compose.yml:69]: Below the env variable `ORGANIZATIONS`

File: server/Dockerfile (hunk 9 above)
Comment (ostefano) [reply server/Dockerfile]: ACK

File: server/files/entrypoint_internal.sh (hunk 8 above)
Comment (ostefano) [reply server/files/entrypoint_internal.sh:217]: Yes

File: server/files/entrypoint_nginx.sh (hunk 7 above)
Comment (ostefano) [reply server/files/entrypoint_nginx.sh:152]: My understanding is that it was redundant (see chmod -R  of line above)

File: docker-compose.yml (hunk 10 above)
Comment (ostefano) [reply docker-compose.yml:69]: Can do, but that would require editing the READMe and documenting them in there.

File: build-docker-compose.yml (hunk 3 above)
Comment (ostefano) [reply build-docker-compose.yml]: Yes warnings are fine

**Comment by ostefano** [comment]
@Phat3 all done, check and approve if happy

**Review by Phat3 — APPROVED** [review]

**Pull request merged**